docs/specs/browserResidentDesktopAgents.md

```diff
@@ -0,0 +1,224 @@
+        connection.appId = appId; // Should be the resolved fully qualified appId
+        connection.instanceId = uuid();
+        connection.appMetadata = new AppMetadata(); // To be set by the DA
+        connection.channel.port1.postMessage({
+            type: "WCPValidateAppIdentityResponse",
+            payload: {
+                desktopAgentDetails: {
+                    agentType: "PARENT",
+                    appId: data.appId,
+                    instanceUUid: connection.instanceUuid,
+                    instanceId: connection.instanceId,
+                },
+                appMetaData: connection.appMetaData,
+                implementationMetadata // The DA's representation of itself as ImplementationMetadata type
+            }
+        }, connection.origin)
+    } else {
+        connection.port.postMessage({
+            type: "WCPValidateAppIdentityResponse",
+            payload: {
+                error: `Origin "${origin}" for "${appId}" didn't match AppD record`
+            }
+        }, connection.origin)
+    }
+}
+```
+
+### Step 3 - DesktopAgent Operations (BCP)
+
+Each message should be responded to with its corresponding response when a response should contain data, or with `BCPAck` if only an acknowledgement is required.
+
+See [Browser Communication Protocol](./browserCommunicationProtocol.md)
+
+See bcp.ts for a full list of BCP messages. 
+
+## Implementing DAs in hidden iframes
+
+DA providers can leverage hidden iframes to establish a communication mechanism that is independent of Parent windows. This approach allows apps to connect to a DA even when they were not opened by that DA.
+
+The hidden iframe url can be provided in two ways:
+
+1) By a Parent DA - This allows DAs to redirect communications to a DA in a hidden iframe at a known location. The main benefit of this approach is that it can allow a system to continue to operate even if the Parent is closed.
+
+2) By a failover function - When no parent DA can be found (such as when a tab is opened directly by an end user) then a failover function can provide a url. In such a scenario, failover is instantaneous because the absence of Parent windows is deterministic. Note that this approach may compromise Write Once Run Anywhere (WORA) unless apps use dynamic mechanisms to determine the hidden url (such as through clues provided in their href.)
+
+The same WCP mechanism is used to connect to DAs located in hidden iframes. Likewise, the same BCP messages are used for communications.
+
+## UI Channel Selector and Intent Resolver
+
+A DA may implement its own Channel Selector and Intent Resolver or may utilize the one provided by "@finos/fdc3" via `getAgent()` ("built-in UI"). For instance, a DA may not have the ability to present a channel selector in a window that has been opened with `window.open()`. The built-in UI can be leveraged in this circumstance. Send the `BCPResolveIntent` and `BCPInitializeChannelSelector` messages to invoke the built-in UI.
```

Comment:
Kind of.   This area probably needs a lot more expansion and POC work.  An App can use a default intent resolver or specify its own.  The default one will use a webpage provided by the DA.  

Channel selection we don't have a POC of yet.